src/mappings/utils.ts

```diff
@@ -95,3 +97,15 @@ export function dateToDayKey(timestamp: any): number {
   const day = date.getUTCDate().toString().padStart(2, "0");
   return parseInt(`${year}${month}${day}`);
 }
+
+export const fetch = async (url: string) => {
+  const response = await crossFetch(url);
+
+  if (response.status >= 400) {
+    const errorText = await response.text();
```

Comment:
What's the purpose of checking the response status? What if it's in the 300 range?

Reply:
Why not use `try/catch`? That way you don't need to check for `response` status on line `104`? If there is an error, the `catch` block can throw an error accordingly.

Reply:
crossFetch wont automatically throw an error in cases where it receives a proper response. such as when we hit this URL https://main-a.api.agoric.net/cosssmos/auth/v1beta1/module_accountss 
We receive a 501 but it doesnt throw an error. in those cases we manually check if we have an error code and throw an error ourselves

Reply:
> What if it's in the 300 range?

3XX are for redirect codes which are not errrors

Reply:
> crossFetch wont automatically throw an error in cases where it receives a proper response. such as when we hit this URL https://main-a.api.agoric.net/cosssmos/auth/v1beta1/module_accountss

Interesting. I reckon this is something wrong with the endpoint we're hitting, right?

Reply:
> 3XX are for redirect codes which are not errrors

Yeah. However, it does not indicate a successful response. Perhaps we should handle it too.

Reply:
> Interesting. I reckon this is something wrong with the endpoint we're hitting, right?
This is the case with most API responses actually. a lot of libraries dont throw an error implicitly and expect you to do it yourself

Reply:
> Yeah. However, it does not indicate a successful response. Perhaps we should handle it too.

300 are successful responses in the sense that we can actually obtain useful responses from them. compared to 400s and 500s which just represent a failure

Reply:
> 300 are successful responses in the sense that we can actually obtain useful responses from them. compared to 400s and 500s which just represent a failure

In that regard, 400 and 500 are also useful responses converting us what's wrong :D

Reply:
@rabi-siddique we look for errors in 400+ range. Here we are just checking for errors in response.